pandas/core/index.py

```diff
@@ -2543,11 +2550,12 @@ def insert(self, loc, item):
         -------
         new_index : Index
         """
-        _self = np.asarray(self)
-        item_idx = Index([item], dtype=self.dtype).values
-        idx = np.concatenate(
-            (_self[:loc], item_idx, _self[loc:]))
-        return Index(idx, name=self.name)
+        indexes=[self[:loc],
+                 Index([item]),
+                 self[loc:]]
+
+        return indexes[0].append(indexes[1]).append(indexes[2])
```

Comment:
sorry, why was it a problem to use ``_shallow_copy`` then? it breaks on ``MultiIndex`` right? That does no infererence, which I don't think you need here anyhow.

This should just work if it doesn't that something small, but subtle is wrong. ``PeriodIndex``has the same mix-ins as all of the other datetimelike.